minkindr_gtsam/src/rotation-quaternion-gtsam.cc

```diff
@@ -75,18 +75,33 @@ EVector3 inverseRotate(const EQuaternion& C, const EVector3& p) {
   return EVector3(&inverse_rotate_point, C, p);
 }
 
+template <typename Scalar_ = double>
+inline bool isLessThanEpsilons4thRoot(Scalar_ x){
+  static const Scalar_ epsilon4thRoot = pow(std::numeric_limits<Scalar_>::epsilon(), 1.0/4.0);
+  return x < epsilon4thRoot;
+}
+
 Eigen::Vector3d rotationLogImplementation(const RotationQuaternion& C,
                                           OptionalJacobian<3, 3> JC) {
   Eigen::Vector3d aa = C.log();
   if(JC) {
     double phi = aa.norm();
-    if(phi < 1e-10) {
+    if(phi == 0) {
       JC->setIdentity();
     } else {
-      double cot = - std::sin(phi)/(std::cos(phi)-1);
-      double a1 = 1/(phi*phi) * (1.0 - 0.5*phi*cot);
+
+      double phiAbs = fabs(phi);
       Eigen::Matrix3d px = kindr::minimal::skewMatrix(aa);
-      (*JC) = Eigen::Matrix3d::Identity() - 0.5 * px + a1 * px * px;
+
+      double a;
+      if(!isLessThanEpsilons4thRoot(phiAbs)) {
```

Comment:
Ah interesting! You try to find the smallest gap between the two? I was more looking at below which phi the polynomial is less off the analytic (true) value than machine precision (so in that range the poly is just good). So if I was roughly right with that we can observe here that they both leave the true value up to 0.01 very similar and get even closer. But this also means that if I was right the upper expression is worse than I thought and we should add the next polynomial term. That would be then of fourth order and should hold up to about the sixth square root of eps.

The fourth order term would be ```phi^4 / 30240```. So in that sequence you could extract ```12 * 60``` and use ```30240 / (12 * 60)  = 42``` ..
Amazing! Who would have guessed that the answer is 42?